tests/cancun/eip1153_tstore/test_tstorage_clear_after_tx.py

```diff
@@ -0,0 +1,96 @@
+"""
+Ethereum Transient Storage EIP Tests
+https://eips.ethereum.org/EIPS/eip-1153
+"""
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Environment,
+    Initcode,
+    Transaction,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .spec import ref_spec_1153
+
+REFERENCE_SPEC_GIT_PATH = ref_spec_1153.git_path
+REFERENCE_SPEC_VERSION = ref_spec_1153.version
+
+
+@pytest.mark.valid_from("Cancun")
+def test_tstore_clear_after_deployment_tx(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+):
```

Comment:
```suggestion
@pytest.mark.valid_from("Cancun")
@pytest.mark.with_all_evm_code_types
def test_tstore_clear_after_deployment_tx(
    blockchain_test: BlockchainTestFiller,
    pre: Alloc,
    evm_code_type: EVMCodeType,
):
```

I feel like we could expand this test a little bit to support EOF by simply adding this marker, and then in the test doing something like:

```python
if evm_code_type == EVMCodeType.EOF_V1:
  code = Container.Init(deploy_container=Container.Code(Op.SSTORE(1, Op.TLOAD(1)) + Op.STOP), initcode_prefix=Op.TSTORE(1, 1))
else:
  code = Initcode(deploy_code=deploy_code, initcode_prefix=init_code)
```
As of now it will not run in EOF until the test is filled in a fork where EOF is active.

Reply:
Have added!